HLRC: Create skeleton code for xpack APIs

hub-cap opened the pull request.

hub-cap commented:
@spinscale just FYI i copy-pasta'd some of your changes into this PR so it can be merged ASAP. If yours merges first then ill gladly remove my changes that include the watcher client and changes to the tests/license.

nik9000 commented on client/rest-high-level/src/main/java/org/elasticsearch/client/XPackClient.java:
```diff
@@ -40,11 +40,47 @@
+        this.watcherClient = new WatcherClient(restHighLevelClient);
     }
 
+    public DeprecationClient getDeprecationClient() { return deprecationClient; }
```
I think it is important that these have Javadoc and that it mention X-Pack.

nik9000 commented on client/rest-high-level/src/main/java/org/elasticsearch/client/SecurityClient.java:
```diff
@@ -0,0 +1,28 @@
+ */
+package org.elasticsearch.client;
+
+public final class SecurityClient {
```
I think these should also have Javadoc and mention X-Pack.

nik9000 commented on client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/MiscellaneousDocumentationIT.java:
```diff
@@ -97,8 +99,8 @@ public void testXPackInfo() throws Exception {
-            assertEquals(XPackInfoResponse.BASIC_SELF_GENERATED_LICENSE_EXPIRATION_MILLIS,
-                    license.getExpiryDate());                          // <3>
+            assertThat(license.getExpiryDate(),
+                is(greaterThan(Instant.now().toEpochMilli())));        // <3>
```
Could you change the prose in the asciidoc file for this callout?

javanna commented on client/rest-high-level/src/main/java/org/elasticsearch/client/MLClient.java:
```diff
@@ -0,0 +1,28 @@
+ */
+package org.elasticsearch.client;
+
+public final class MLClient {
```
not sure, but should this be MachineLearningClient?

hub-cap replied:
I considered this, as the IT is MLIT which is the nastiest name ever. But the spec shows them as `.ml.` so I wanted to stay true to it.. Do you consider renaming the class diverging, or is this ok? I could still do a `ml()` getter in the XPackClient so the `hlrc.xpack().ml()` would still work out.

Also, wow, i cant believe i did this... i need to remove the get's off those methods so they are not `hlrc.xpack().getML()` !!!!!

javanna replied:
good point about the get prefix, I had missed it. I don't mind leaving naming as-is.

javanna commented on client/rest-high-level/src/test/java/org/elasticsearch/client/MLIT.java:
```diff
@@ -0,0 +1,22 @@
+ */
+package org.elasticsearch.client;
+
+public class MLIT extends ESRestHighLevelClientTestCase {
```
the name here seems a bit too concise as well :) Maybe MachineLearningIT ?

hub-cap commented:
* Cleaned up the client getters to do only things like `.watcher()`, `.ml()`, etc. 
* Removed some classes which are not in the xpack completeness issue.
* javadoc'd all the things
* called out the trial and basic license info in the asciidoc.

@javanna or @nik9000 can you eyeball again?

hub-cap commented:
Closing as this info / code is out of date.
